controllers/controllers/resource/template.go

```diff
@@ -71,10 +72,15 @@ func (r *VsphereTemplate) TemplateResources(ctx context.Context, eksaCluster *an
 }
 
 func generateTemplateResources(builder providers.TemplateBuilder, clusterSpec *cluster.Spec, buildOptions ...providers.BuildMapOption) ([]*unstructured.Unstructured, error) {
-	content, err := builder.GenerateDeploymentFile(clusterSpec, buildOptions...)
+	cp, err := builder.GenerateClusterApiSpecCP(clusterSpec, buildOptions...)
 	if err != nil {
 		return nil, err
 	}
+	md, err := builder.GenerateClusterApiSpecMD(clusterSpec, buildOptions...)
```

Comment:
The function is `GenerateCapiSpecMD`, it's mainly about generating CAPI machine deployment. CAPI does not have worker node groups. I feel it is more proper to keep the MD in this case.

Reply:
I guess the way I thought about it was "generate all cluster api spec that is related to worker nodes" rather than specifically for machine deployments. Cp makes sense above because control plane is more generic and is something that we follow too. By naming it worker, we are just letting this function contain all the context as to what constitutes a worker template, which in our case is MachineDeployment related templates and anything else we introduce in the future.

Reply:
The way I think abt it is the function itself is generating CAPI spec. It does not contain our EKS-A business logic. Also all the configs in the template-md are CAPI MD related.